runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/BatchStatefulParDoOverrides.java

```diff
@@ -247,6 +284,21 @@ public void processElement(final ProcessContext c, final BoundedWindow window) {
   }
 
   /**
+   * A key preserving {@link DoFn} that expands the output of a GBK {@code KV<K, Iterable<V>>} into
+   * individual KVs.
+   */
+  static class ExpandGbkFn<K, V>
```

Comment:
I almost want to move this to Flatten, e.g. `Flatten.valueIterables()`, but I think that generally runner authors are the only ones who would use it.

However, it could find a home in `runners-core-construction` if not in the core SDK.

Reply:
I'm sure once we have key preserving as a piece of info available, we will want to expose these for re-use.